src/main/java/com/cburch/logisim/std/tcl/TclComponent.java

```diff
@@ -266,7 +266,7 @@ void getPortsFromServer(InstanceState state, TclComponentData tclComponentData)
         && server_response.length() > 0
         && !server_response.equals("sync")) {
 
-      String[] parameters = server_response.split("\\:");
+      String[] parameters = server_response.split(":");
```

Comment:
It's not TCL related; the split() method in Java takes a Regex, and this was passing the regex `\:`, but `:` is not a special character in regular expressions so the backslash can be dropped. In any case this was not splitting the string with the *raw string* `\:`

Reply:
See this snippet https://ideone.com/c27Y2D

Reply:
Yes, I figured this was the reason you removed `\\` from the Regex. I made the comment just, because I am not sure how the server responses look like, i.e., whether the parameters are separated by `:` or `\\:`. (I would expect only `:`.)